### ISSUE TITLE
Atmosphere framework initalization fails in OSGI environment

### ISSUE THREAD

**Issue by arotnov** [Atmosphere/atmosphere#2324]
The problem is that Atmosphere produces Injectable implementations and consumes it itself by ServiceLoader.
ServiceLoader by default uses current thread classloader, and in OSGI current classloader might not know about implementations of Injectable services (example, vaadin-server bundle trying to instatiate framework and initialization fails, because classes loads by vaadin bundle classloader).
To work ServiceLoader correctly in OSGI environment there is special extending mechanizm, described in OSGI Spec. It could be enabled by adding requrements and capabilities in manifest. For Atmosphere framework they are:
```
Provide-Capability: osgi.serviceloader;osgi.serviceloader="javax.servlet.ServletContainerInitializer",osgi.serviceloader;osgi.serviceloader="org.atmosphere.inject.CDIProducer",osgi.serviceloader;osgi.serviceloader="org.atmosphere.inject.Injectable"
Require-Capability: osgi.extender;filter:="(osgi.extender=osgi.serviceloader.processor)",osgi.extender;filter:="(osgi.extender=osgi.serviceloader.registrar),osgi.serviceloader;filter:="(osgi.serviceloader=org.atmosphere.inject.Injectable)";cardinality:=multiple
```
or by maven-bundle-plugin configuration:
```
<configuration>
    <instructions>
        <Require-Capability>
            osgi.extender;filter:="(osgi.extender=osgi.serviceloader.processor)",
            osgi.extender;filter:="(osgi.extender=osgi.serviceloader.registrar)",
            osgi.serviceloader;filter:="(osgi.serviceloader=org.atmosphere.inject.Injectable)";cardinality:=multiple
        </Require-Capability>
        <Provide-Capability>
            osgi.serviceloader;osgi.serviceloader=javax.servlet.ServletContainerInitializer,
            osgi.serviceloader;osgi.serviceloader=org.atmosphere.inject.CDIProducer,
            osgi.serviceloader;osgi.serviceloader=org.atmosphere.inject.Injectable
        </Provide-Capability>
    </instructions>
</configuration>
```
and of course, package of implementations (org.atmosphere.inject) must be imported by consumer, i.e. by atmosphere-runtime itself :)

**Comment by jfarcand** [comment]
@arotnov Pull request more than welcomed...I agree it has been awhile since I've looked at OSGI. If you have the time, fix it and I will integrate it. Thanks

**Comment by jfarcand** [comment]
Fixed by #2324
